arrow.sh

```diff
@@ -101,8 +104,9 @@ cmake ./src_tmp/cpp
       -DARROW_TENSORFLOW=ON                                                                         \
       -DARROW_GANDIVA=ON                                                                            \
       -DARROW_COMPUTE=ON                                                                            \
+      -DARROW_BUILD_STATIC=OFF                                                                      \
       -DCMAKE_INSTALL_RPATH_USE_LINK_PATH=ON                                                        \
-      -DCLANG_EXECUTABLE=${CLANG_ROOT}/bin-safe/clang
```

Comment:
just wondering, was there a particular reason we used bin-safe/clang here before? Perhaps that was to avoid a collision with the system clang? If so, perhaps we should keep it as is?

Reply:
We switched off the compilation of Clang on MacOSX and used the system one on both osx_x86-64 and osx_arm64, while for the platforms where we compile Clang, we use bin-safe. The reason is that Apple has its own patches and the git version does not compile.

Reply:
ok, and are we sure a system clang with a lower version will not be picked up for arrow?

Reply:
Concerning the issue with "fatal: not a git repository..." I suspect something is wrong in the local directory. Did we run out of space there?

Reply:
It will fail if the version is below 12.0 - there is a check in arrow.

Reply:
hm, ok, but what with a system installation of clang 13 / clang 14 dev versions? (which I have). Probably arrow will not pick them up anyway I guess, since they usually limit the supported clang version. ok, anyway, if this breaks something, we can still revert to specifying the alibuild clang.